### DISCUSSION TITLE
Force blur/pixellation for all images on specific websites as fallback

### DISCUSSION THREAD

**Discussion by libertosurf** [silveredgold/beta-protection#59] [Ideas]
Hello,

So I've been playing with this extension for a bit, and I noticed that sometimes the AI doesn't censor as expected. This is especially the case on reddit (the thumbnails, maybe because they are too small to be processed correctly), but it's also the case if I directly go on a porn site (pornhub for instance).

So, I was wondering if it could be possible to force the blurring/pixellation/etc for specific websites, with a smarter-ish algorithm that goes like:

1. use AI to censor
2. If nothing is censored, then apply blur/pixel/whatever you choose, as a fallback.

I originally thought that's what the "forced domain sites" was about, but as far as I understand it now, it just overrides the "on demand"/"enabled" thing.

Edit: another idea maybe: be able to adjust the settings so that it's less strict on body part detection so that it censors more things even if there are false positives

**Comment by silveredgold** [comment]
To get the easier one out of the way: You can adjust the per-body-part detection sensitivity in the backend if you're using Beta Censoring. This allows you to up the detection threshold meaning it will censor more aggressively, potentially picking up more missed parts at the cost of potentially censoring things unnecessarily.

---

As for the larger point of allowing a blur-as-fallback, I actually experimented with this but it has one big problem: the extension has no way of knowing what's an actual photo/image and what's a UI element masquerading as a button. To use Reddit as an example, the upvote/downvote arrows, mail/inbox/chat icons, the text post icons, awards, snoo icons, and expand/collapse icons are all technically images. In the scenario you described, all of those would get blurred or pixelated since they would not be censored.

That gets even worse on pages that use images for things like button backgrounds, titles, or navigation items as they become largely unusable when blurred or pixelated. Any measure we use to try and "skip" UI elements based on size/tag name/image source also results in more uncensored images like thumbnails sneaking through.
